<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1424</width>
    <height>711</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QScrollArea" name="scrollArea">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>891</width>
      <height>631</height>
     </rect>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>889</width>
       <height>629</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>480</y>
      <width>81</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Agent</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>20</y>
      <width>91</width>
      <height>30</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
    </property>
    <property name="text">
     <string>Scene</string>
    </property>
    <property name="margin">
     <number>-1</number>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_10">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>140</y>
      <width>211</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Default Generate</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>180</y>
      <width>471</width>
      <height>251</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>R/G Generate</string>
     </attribute>
     <widget class="QLineEdit" name="lineEdit_10">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>50</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>10</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Number</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>90</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Y</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QRadioButton" name="radioButton_3">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>14</y>
        <width>112</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Eillpse</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>X</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QRadioButton" name="radioButton_4">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>14</y>
        <width>112</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Circle</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_11">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>10</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>10</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_19">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Length</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_11">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>170</y>
        <width>211</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Random Generate</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>50</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Interval</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_12">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>90</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>500</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_13">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>90</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>200</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_14">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>50</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>2</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_15">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>130</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>50</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_21">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>130</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Width</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_12">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>211</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Grid Generate</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_16">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>130</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>100</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_22">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Radius</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>item Generate</string>
     </attribute>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>50</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Y</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>X</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Length</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_4">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>50</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>500</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_7">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>50</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>200</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_3">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>10</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>2</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_8">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>90</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>50</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>90</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Width</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_8">
      <property name="geometry">
       <rect>
        <x>125</x>
        <y>170</y>
        <width>211</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Item Generate</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_5">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>90</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>100</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Radius</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>10</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Angle</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_17">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>10</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>200</string>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBox_3">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>130</y>
        <width>331</width>
        <height>30</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Circle</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Box</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Rectangle</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_23">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>91</width>
        <height>30</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
      </property>
      <property name="text">
       <string>Shape</string>
      </property>
      <property name="margin">
       <number>-1</number>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>520</y>
      <width>91</width>
      <height>30</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
    </property>
    <property name="text">
     <string>Model</string>
    </property>
    <property name="margin">
     <number>-1</number>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>480</y>
      <width>141</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Generate Region</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>100</y>
      <width>131</width>
      <height>30</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
    </property>
    <property name="text">
     <string>Default Generate</string>
    </property>
    <property name="margin">
     <number>-1</number>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>60</y>
      <width>471</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_2">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>560</y>
      <width>471</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>480</y>
      <width>81</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Wall</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>440</y>
      <width>131</width>
      <height>30</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{

     border-width: 1px;

     border-style: solid;

     border-color: rgb(96, 96, 96);

}</string>
    </property>
    <property name="text">
     <string>Manual Generate</string>
    </property>
    <property name="margin">
     <number>-1</number>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_5">
    <property name="geometry">
     <rect>
      <x>1260</x>
      <y>480</y>
      <width>141</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Safe Region</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_13">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>610</y>
      <width>211</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Confirm</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_14">
    <property name="geometry">
     <rect>
      <x>1170</x>
      <y>610</y>
      <width>211</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Cancel</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1424</width>
     <height>28</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
